### PR TITLE
remove death code

### PR DESCRIPTION
**Pull request by siscia**
Remove death code.

Everything seems to compile just fine without.

### CODE REVIEW

File: cvmfs/compat.h [hunk 1]
```diff
@@ -647,10 +647,6 @@ class StringHeap : public SingleCopy {
  private:
   void AddBin(const uint64_t size) { assert(false); }
 
-  uint64_t size_;
```
Comment (jblomer) [inline cvmfs/compat.h]: This is in fact not dead code.  It's necessary for the hot patch in order to establish the memory layout of that class in a previous version of cvmfs.

Comment (siscia) [reply cvmfs/compat.h]: All the file or just `size_` ?

Comment (jblomer) [reply cvmfs/compat.h]: All of them

Comment (siscia) [reply cvmfs/compat.h]: Removed that file!

**Pull request merged**